packages/tpetra/core/test/MatrixMatrix/MatrixMatrixKernels_UnitTests.cpp

```diff
@@ -166,7 +166,16 @@ getIdentityMatrixWithMap (Teuchos::FancyOStream& out,
   return identityMatrix;
 }
 
-
+template<class Matrix_t>
+RCP<Matrix_t>
+copyMatrix(RCP<Matrix_t> &A) {
```

Comment:
If you don't intend on the RCP being an output argument, then pass it either by value `RCP<Matrix_t>` or by const reference `const RCP<Matrix_t>&`.

Reply:
More importantly, if you need to deep-copy a CrsMatrix a lot, then we should add a new CrsMatrix deep-copy constructor like `MultiVector<> Y (X, Teuchos::Copy)`.

Reply:
As discussed.